### ISSUE TITLE
Foreign Key support on PlanetScale

### ISSUE THREAD

**Issue by janpio** [prisma/prisma#22266]
PlanetScale just release foreign key support: https://planetscale.com/blog/announcing-foreign-key-constraints-support That could mean the [`relationMode="prisma"`](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/relation-mode) is not needed any more to use Prisma with a database hosted on PlanetScale.

For now it is a beta feature on their platform that you have to opt-in on a database level. We have tested Prisma on PlanetScale and on Vitess, the underlying technology, and are pretty sure this will be a pretty smooth transition.

Anyway, we suggest you first try it with a copy of your database and see how your application using Prisma Client behaves.

If you encounter any unexpected behavior, please reply in this issue or open a new bug issue and link it here. Thank you!

**Comment by janpio** [comment]
First possible problem reported at PlanetScale discussions: https://github.com/planetscale/discussion/discussions/569#discussion-5928257

**Comment by serranoarevalo** [comment]
FWIW Foreign Key Constraints are out of Beta an Generally Available.

https://planetscale.com/blog/foreign-key-constraints-are-now-generally-available

Does this also mean we don't have to manually create indexes anymore?

**Comment by janpio** [comment]
Correct, we have updated our documentation to take that into account: https://www.prisma.io/docs/orm/overview/databases/planetscale As the default is not changed yet, we for now stay with our `relationMode=prisma` as the default as well and will depend on the feedback here to inform that for the future.

**Comment by ahkhanjani** [comment]
Hey @janpio. Unfortunately there's a big downside to `relationMode=prisma` waiting to get fixed: #16390
